lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -30,6 +31,67 @@ def post_sql_statements( table_name, options ) # :nodoc:
     end
   end
 
+  # Add a column to be updated on duplicate key update
+  def add_column_for_on_duplicate_key_update( column, options={} ) # :nodoc:
+    if options[:on_duplicate_key_update].is_a?( Hash )
+      columns = options[:on_duplicate_key_update].fetch(:columns, [])
+      case columns
+      when Array then columns << column.to_sym unless columns.include?( column.to_sym )
+      when Hash then columns[column.to_sym] = column.to_sym
+      end
+    end
+  end
+
+  # Returns a generated ON CONFLICT DO UPDATE statement given the passed
+  # in +options+.
+  def sql_for_on_duplicate_key_update( table_name, options={} ) # :nodoc:
+    return unless options.is_a?( Hash )
+
+    key = Array( options[:key] ).join( ', ' )
+    raise ArgumentError, 'Attribute :key not found' if key.empty?
```

Comment:
Yes, that's correct. The ON CONFLICT clause requires a conflict_target, unless the conflict_action is DO NOTHING. Currently the only time that conflict_action would be DO NOTHING is if :columns is an empty array *and* the table does not have update timestamp columns that automatically get assigned. I can update the code to make :key optional in that scenario

Unfortunately, as far as I can tell, this requirement makes matching the MySQL interface not possible :frowning: